sdk/python/feast/diff/registry_diff.py

```diff
@@ -129,8 +132,10 @@ def diff_registry_objects(
     current_spec: FeastObjectSpecProto
     new_spec: FeastObjectSpecProto
     if isinstance(
-        current_proto, (DataSourceProto, ValidationReferenceProto)
-    ) or isinstance(new_proto, (DataSourceProto, ValidationReferenceProto)):
+        current_proto, (DataSourceProto, ValidationReferenceProto, PermissionProto)
```

Comment:
are you sure we need this casting for Permissions? (just asking, I imagine this is done to handle sub-classes, as DataSource is abstract, but not completely sure about that)

Reply:
Not sure consulting with Tornike regarding the exisiting code, waiting for his input on this, will mark it as resolved for now, will change it in the future if its needed